test/unit/source/tile.test.js

```diff
@@ -263,6 +264,27 @@ test('expiring tiles', (t) => {
     t.end();
 });
 
+test('rtl text detection', (t) => {
+    t.test('Tile#hasRTLText is true when a tile loads a symbol bucket with rtl text', (t) => {
+        const tile = new Tile(new OverscaledTileID(1, 0, 1, 1, 1));
+        const symbolBucket = createSymbolBucket('test', 'Test', 'مرحبا', new CollisionBoxArray());
+        symbolBucket.hasRTLText = true;
```

Comment:
Why do we have to manually set `symbolBucket.hasRTLText = true`? Shouldn't that happen automatically like in the symbol bucket tests above? It seems like this test is kind of trivial if we manually set this to `true` since in `tile.js`, it basically just is `this.hasRTLText = symbolBucket.hasRTLText`.

Reply:
ahh, its because `hasRTLText` can only be determined after bucket.populate has been called.
I'm only using `createSymbolBucket` to create a stub of a symbol bucket here. I've added a comment to explain.

Reply:
There's a couple typos in the comment